Sources/TemplateCollection.swift

```diff
@@ -0,0 +1,51 @@
+public enum TemplateCollectionError: Error {
+    case noSuchTemplate(named: String)
+}
+
+public struct TemplateCollection {
+    public var templates: [String:Template]
+
+    public init(templates: [String:Template] = [:]) {
+        self.templates = templates
+    }
+
+    public func get(template name: String) throws -> Template {
+        guard let template = templates[name] else {
+            throw TemplateCollectionError.noSuchTemplate(named: name)
+        }
+        return template
+    }
+
+    public func render(template name: String, with context: Context = .array([])) throws -> String {
+        return try get(template: name).render(with: context, partials: self.templates)
+    }
+}
+
+// MARK: IO
+import Foundation
+extension TemplateCollection {
+    public init(basePath: String = FileManager.default.currentDirectoryPath, directory: String) throws {
+        let path = basePath + directory
+        let files = try FileManager.default.contentsOfDirectory(atPath: path)
+
+        var templates = [String:Template]()
+
+        for file in files where file.split(separator: ".").last == "mustache" {
```

Comment:
also.. `Foundation` has a property to get the file extension.. I think it's on `NSString` IIRC